packages/popover/src/Popover/Popover.tsx

```diff
@@ -160,14 +175,37 @@ export const Popover = forwardRef<HTMLDivElement, PopoverComponentProps>(
       return children;
     };
 
+    useEffect(() => {
+      if (renderMode !== RenderMode.TopLayer || !onToggle) {
+        return;
+      }
+
+      // @ts-expect-error - `toggle` event not supported pre-typescript v5
+      refs.floating.current?.addEventListener('toggle', onToggle);
+      return () =>
+        // @ts-expect-error - `toggle` event not supported pre-typescript v5
+        refs.floating.current?.removeEventListener('toggle', onToggle);
+    }, [onToggle, renderMode]);
+
+    useEffect(() => {
+      if (renderMode !== RenderMode.TopLayer) {
+        return;
+      }
+
+      if (context.open) {
+        // @ts-expect-error - Popover API not currently supported in react v18 https://github.com/facebook/react/pull/27981
+        refs.floating.current?.showPopover?.();
+      } else {
+        // @ts-expect-error - Popover API not currently supported in react v18 https://github.com/facebook/react/pull/27981
+        refs.floating.current?.hidePopover?.();
+      }
```

Comment:
nit: might want to check whether `context.open` has changed, otherwise we'll be calling show/hide unnecessarily. Shouldn't be a big deal but idk how this API works under the hood